### PR TITLE
feat!: drop moment lib date format

### LINKED ISSUE #428: Replace `moment` with `luxon`
https://github.com/fastify/fast-json-stringify/blob/03f8714fc2412c3e46cb7bbb947e316cad0681bc/package.json#L42

`moment` is considered to be in "maintenance mode" https://momentjs.com/docs/#/-project-status/
> We now generally consider Moment to be a legacy project in maintenance mode. It is not dead, but it is indeed done.

Until `Temporal` is available, `moment` should be switched out for `luxon` https://moment.github.io/luxon/#/

### PR DESCRIPTION
**Pull request by ivan-tymoshenko**
❗ BREAKING CHANGE

Motivation: https://github.com/fastify/fast-json-stringify/pull/441#discussion_r884803343

Close #428

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
I'm ok with the change. I'll likely bump the major at the end of this week. If there are other significant changes, let us know.

**Comment by ivan-tymoshenko** [comment]
@mcollina I will try to push everything that I have, but for now, I think there is a lot of work to do. I won't make it by the end of this week. I don't say that we shouldn't release it, just want to let you know it, maybe it would be better to wait another with a major release.

**Comment by mcollina** [comment]
I have an hard date for the fastify v4 release, that's all.

File: package.json [hunk 1]
```diff
@@ -39,7 +39,7 @@
     "compile-json-stringify": "^0.1.2",
     "inquirer": "^8.2.4",
     "is-my-json-valid": "^2.20.0",
-    "moment": "^2.24.0",
+    "moment": "^2.29.3",
```
Comment (jsumners) [inline package.json:42]: Should this be removed?

**Review by jsumners — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (mcollina) [reply package.json:42]: I think so

Comment (ivan-tymoshenko) [reply package.json:42]: We still use it for tests. I moved it to devDep, but I guess we can replace it with `luxon` there. https://github.com/fastify/fast-json-stringify/issues/428

**Pull request merged**